paasta_tools/cli/cmds/validate.py

```diff
@@ -417,6 +421,58 @@ def validate_autoscaling_configs(service_path):
     return returncode
 
 
+def validate_secrets(service_path):
+    soa_dir, service = path_to_soa_dir_service(service_path)
+    system_paasta_config = load_system_paasta_config()
+    return_value = True
+    for cluster in list_clusters(service, soa_dir):
+        vault_env = system_paasta_config.get_vault_cluster_config().get(cluster)
+        if not vault_env:
+            paasta_print(failure(f"{cluster} not found on vault_cluster_map", ""))
+            return_value = False
+            continue
+
+        for instance in list_all_instances_for_service(
+            service=service, clusters=[cluster], soa_dir=soa_dir
+        ):
+            instance_config = get_instance_config(
+                service=service,
+                instance=instance,
+                cluster=cluster,
+                load_deployments=False,
+                soa_dir=soa_dir,
+            )
+            for env_value in instance_config.config_dict.get("env", {}).values():
+                if is_secret_ref(env_value):
```

Comment:
probably worth taking this block out into separate function - easier to understand what's going on and easier to test

Reply:
sounds good. I'm going to extract some logic into a helper function.